### PR TITLE
Refactor interrupts management

### PR DESCRIPTION
**Pull request by n1tram1**
InterruptManagement is not very useful (it is not used, and its ergonomics are not great).

1. In my opinion `ArchitectureInterrupts` isn't very usefull, the only useful method in it is to enable interrupts, but then shouldn't that just be in the `Architecture` trait.
2. Setting the timer, probably should be moved to a driver and have it enable an IRQ line with its handler (do all architectures have the timer directly as part of the architecture ?).
3. If ALL architectures have a timer, can't we put in the `Architecture` trait (for example I know that on ARMv8, timer don't have to generate an interrupt, they can be used via polling instead)

### CODE REVIEW

**Comment by n1tram1** [comment]
I know, I know, nothing is implemented for rv64

**Comment by n1tram1** [comment]
Evne though CI passes, this isn't done, the `irq::IrqManager` trait should be implemented for the plic too. (@Skallwar :eyes: )

File: kernel/src/riscv64_qemuvirt_main.rs [hunk 1]
```diff
@@ -74,7 +74,7 @@ extern "C" fn k_main(_core_id: usize, device_tree_ptr: usize) -> ! {
 
     kprintln!("[OK] Setup virtual memory");
 
-    let mut arch_interrupts = InterruptsImpl{};
+    let mut arch_interrupts = InterruptsImpl {};
```
Comment (CohenArthur) [inline kernel/src/riscv64_qemuvirt_main.rs:77]:
```suggestion
    let mut arch_interrupts = InterruptsImpl;
```

We can have empty structs, people! Stop putting extra `{}` at the end of them :D This isn't C!

And yeah, that's not relevant to this PR and the comment would be better adressed on the struct definition itself. But still!

**Review by CohenArthur — APPROVED** [review]

**Comment by Skallwar** [comment]
Should this PR not be on top of `implement-aarch64-exceptions` instead of main?

**Comment by n1tram1** [comment]
Done in https://github.com/ks0n/goOSe/pull/135

**Pull request closed without merging**